### DISCUSSION TITLE
@teleport is being shown on the view

### DISCUSSION THREAD

**Discussion by jacksontong** [livewire/livewire#6916] [7. Bugs (archived)]
### Livewire version

3.0.5

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

I'm using modal of Jetstream and when I deploy on the server it shows like this
![image](https://github.com/livewire/livewire/assets/7871218/6120ef55-814f-4417-b331-8f1f3b4958c2)

I find that because I run `php artisan view:cache`, but if I clear the view then it's gone.

```php
@teleport('body')
    <x-modal id="create-accountant-modal" wire:model="showModal">

@endteleport
```

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by TwilightDuck** [comment]
You're not closing your `<x-modal>` tag.

**Reply by jacksontong** [reply]
That's not the reason why it's not working with `view:cache`, it's just an example without full code. I know why it's not working with `view:cache`

https://github.com/livewire/livewire/blob/main/src/Features/SupportTeleporting/SupportTeleporting.php

```php
app('livewire')->directive('teleport', function ($expression) {
    return '<template x-teleport="<?php echo e('.$expression.'); ?>">';
});
```
This is how Livewire register the `teleport` directive, and behind the scene it calls to `app(ExtendBlade::class)->livewireOnlyDirective($name, $callback);` https://github.com/livewire/livewire/blob/main/src/LivewireManager.php#L47

And `view:cache` only compile for their own blade https://github.com/laravel/framework/blob/10.x/src/Illuminate/Foundation/Console/ViewCacheCommand.php#L59

I find `teleport` is useless if it's not working with `view:cache` I have to remove it from my code.

**Comment by Mesuva** [comment]
I'm hitting this quite often, on 3.0.7

I've not been able to work out an exact pattern behind when it happens and when it resolves. 
But I agree it's something to do with view:cache being run.

As a workaround, swapping out the blade directives with their replacements seems to work fine:

e.g swapping out:
```
@teleport('#loading-spinner')
...
@endteleport
```

for something l

```
<template x-teleport='#loading-spinner'>
...
</template> 
```

So the Alpine functionality is working fine, it's just the directive that seems to be failing to do the find-and-replace in the view templates correctly.

**Comment by calebdw** [comment]
This is still an issue on 3.1.0---caching the views (which we always do in production) breaks the teleport

**Comment by jonhassall** [comment]
Happened today for me with caching views. @calebdw's solution resolved it. v3.4.4

**Comment by robsontenorio** [comment]
The solution is not to cache views?

**Comment by jonhassall** [comment]
> The solution is not to cache views?

My mistake. @Mesuva's workaround of swapping from Blade to the Alpine tags worked for me.

**Comment by gbalduzzi** [comment]
I can confirm the problem is still present in v3.4.9

I think the teleport docs should at the very least mention this issue. Unless you find this discussion it's very hard to understand what is going on

**Reply by gbalduzzi** [reply]
My bad, apparently #8282 fixed the problem and it shipped in v3.4.11
